modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/BaseUserAccountResourceImpl.java

```diff
@@ -1741,9 +1742,41 @@ public void create(
 		}
 
 		if ("UPSERT".equalsIgnoreCase(createStrategy)) {
-			userAccountUnsafeConsumer =
-				userAccount -> putUserAccountByExternalReferenceCode(
-					userAccount.getExternalReferenceCode(), userAccount);
+			String updateStrategy = (String)parameters.getOrDefault(
+				"updateStrategy", "UPDATE");
+
+			if ("UPDATE".equals(updateStrategy)) {
+				userAccountUnsafeConsumer =
+					userAccount -> putUserAccountByExternalReferenceCode(
+						userAccount.getExternalReferenceCode(), userAccount);
+			}
+
+			if ("PARTIAL_UPDATE".equals(updateStrategy)) {
+				userAccountUnsafeConsumer = userAccount -> {
+					try {
+						UserAccount getUserAccount =
+							getUserAccountByExternalReferenceCode(
+								userAccount.getExternalReferenceCode());
+
+						patchUserAccount(
+							getUserAccount.getId() != null ?
+								getUserAccount.getId() :
+									_parseLong(
+										(String)parameters.get(
+											"userAccountId")),
+							userAccount);
+					}
+					catch (NoSuchModelException noSuchModelException) {
+						postUserAccount(userAccount);
```

Comment:
I think in this case if the `parameters` map contains `accountId`, we would invoke two different methods:

- postUserAccount
- postAccountUserAccount

Reply:
Maybe we need to manage that with a UnsafeConsumer as in the `INSERT` `createStrategy` section

Reply:
Just to see if understand this case correctly:

- If a user submits **accountId** through configuration, then all records from the import file will be treated as UserAccounts of an Account identified by accountId.

- Otherwise just create a new UserAccount?

Reply:
Yeah. Let's imagine a developer wants to import a collection of non-existing `UserAccount` associated to a given `accountId` with `createStrategy` == `UPSERT` and `updateStrategy` == `PARTIAL_UPDATE`:

```
userAccountUnsafeConsumer = userAccount -> {
	try {
		UserAccount getUserAccount =
			getUserAccountByExternalReferenceCode(
				userAccount.getExternalReferenceCode());

		patchUserAccount(
			getUserAccount.getId() != null ?
				getUserAccount.getId() :
					_parseLong(
						(String)parameters.get(
							"userAccountId")),
			userAccount);
	}
	catch (NoSuchModelException noSuchModelException) {
		postUserAccount(userAccount);

		if (parameters.containsKey("accountId")) {
			postAccountUserAccount(
				_parseLong((String)parameters.get("accountId")),
				userAccount);
		}
	}
};
```

So, for each UserAccount: the `getUserAccountByExternalReferenceCode` method would throw an exception, the `postUserAccount` method would be executed, a UserAccount would be created (without assigning it to an Account), and then, the `postAccountUserAccount` would be executed too, but the UserAccount already exists! Probably an exception would be thrown